python/social_bot/gazebo_agent.py

```diff
@@ -180,10 +195,37 @@ def get_full_states_observation(self, teacher):
         """
         task_specific_ob = teacher.get_task_specific_observation(self)
         agent_pose = np.array(self.get_pose()).flatten()
-        agent_vel = np.array(self.get_velocities()).flatten()
-        internal_states = self.get_internal_states()
-        obs = np.concatenate(
-            (task_specific_ob, agent_pose, agent_vel, internal_states), axis=0)
+        if self._use_simple_full_states:
```

Comment:
This seems to be dedicated to GoalTask. Maybe move these to GoalTask.specific_observation() ?

I thought agent_pose and agent_vel (including position and rotation) are commonly used before so I put them here. But it seems many task don't need all of them. Perhaps we can remove this "get_full_states_observation" function, directly use "get_task_specific_observation" instead. Let the tasks define all there observations for the non-image case. Both simpler and more flexible.

Reply:
I think this code should still live here, as it's just changing from a global coordinate system to an egocentric one.  It could help tasks like pick and place as well?

Reply:
Different tasks have different task_specific_ob. It's hard to be compatible for all of them. Codes like 
```
x = task_specific_ob[0] - agent_pose[0]
task_specific_ob = task_specific_ob[3:]
```
will not work with other tasks since there might be touch sensors, rotations, angles, pos only contains x and y axis, etc.

So I still think that we should move them to GoalTask.specific_observation. For egocentric coordinate, we can add a common helper function to convert a global position to egocentric one given agent pose.

Reply:
I see, good catch Jiangtao!  I moved logic to taskspecific obs, and left only the flags in the GazeboAgent, which I think is a natural place for them to live.

Let me know how you like this way.

Reply:
Let's move the flag to task, and rename to something like use_egocentric_states. I think configure the observation egocentric or not in task is also very natural.

Reply:
agent_pose and agent_vel (including position and rotation) should be removed. That bring redundant or unused dimensions in some other task's observation. So remove "get_full_states_observation" directly use "get_task_specific_observation" would be better. Let the tasks define all there observations for the non-image case. Both simpler and more flexible.

This require to remove some redundant or unused dimensions for other task. Let me do this.

Reply:
Sounds good.  Thanks!